lib/client.js

```diff
@@ -335,22 +338,7 @@ class Client extends EventEmitter {
         return
       }
 
-      let reset = false
-
-      if (body) {
-        // Some servers disconnect requests with body even after
-        // successful completion. Assume this always happens and
-        // explicitly reset the connection once response has been
-        // received. Queue will automatically resume once reconnected.
-
-        // TODO: This is too strict. Try to figure out when/how/why some
-        // servers do this.
-
-        reset = true
-        this[kQueue].pause()
-      }
```

Comment:
This is actually no needed. The problem we were seeing was because of message interleaving since we weren't pausing the queue when piping request body.